### PR TITLE
Fix empty LR scheduler for old jobs, clean up argument validation

### PR DESCRIPTION
**Pull request by mryab**
The introduction of `Literal` in https://github.com/togethercomputer/together-python/pull/273 broke parsing of old jobs without the learning rate scheduler specified, as Pydantic receives an empty string for the type and can't match it to any of the LR scheduler types that we have defined. 

The error looks like this:
```
data.4.lr_scheduler.FinetuneCosineLRScheduler.lr_scheduler_type
  Input should be 'cosine' [type=literal_error, input_value='', input_type=str]
    For further information visit https://errors.pydantic.dev/2.10/v/literal_error
```

As a simple fix before any backend changes, this PR adds an empty scheduler type and enables it **only** in `FinetuneResponse`. `together fine-tuning list` works with the new fix

The PR also makes a few cleanups around the codebase related to LR scheduling and fixes a hidden bug introduced in the previous PR, where `lr_scheduler_args` was renamed to `lr_scheduler` but Pydantic silently accepted the previous name of the field

### CODE REVIEW

File: src/together/resources/finetune.py [hunk 1]
```diff
@@ -152,21 +152,19 @@ def createFinetuneRequest(
...
-    # Default to generic lr scheduler
-    lrScheduler: FinetuneLRScheduler = FinetuneLRScheduler(lr_scheduler_type="linear")
-
+    lr_scheduler: FinetuneLRScheduler
     if lr_scheduler_type == "cosine":
         if scheduler_num_cycles <= 0.0:
             raise ValueError("Number of cycles should be greater than 0")
 
-        lrScheduler = FinetuneCosineLRScheduler(
-            lr_scheduler_args=FinetuneCosineLRSchedulerArgs(
```
Comment (mryab) [inline src/together/resources/finetune.py:163]: As you can see here, we passed `lr_scheduler_args` despite this field not being defined for `FinetuneCosineLRScheduler`

File: src/together/types/finetune.py [hunk 2]
```diff
@@ -358,14 +358,20 @@ class FinetuneLRScheduler(BaseModel):
...
 
-class FinetuneCosineLRScheduler(FinetuneLRScheduler):
+class CosineLRScheduler(FinetuneLRScheduler):
     lr_scheduler_type: Literal["cosine"] = "cosine"
-    lr_scheduler: FinetuneCosineLRSchedulerArgs | None = None
+    lr_scheduler_args: CosineLRSchedulerArgs | None = None
+
+
+# placeholder for old fine-tuning jobs with no lr_scheduler_type specified
+class EmptyLRScheduler(FinetuneLRScheduler):
```
Comment (thepowerfuldeez) [inline src/together/types/finetune.py:372]: Smart! Looks good to me if we have no other options to support old jobs

**Review by thepowerfuldeez — APPROVED** [review]

**Review by azahed98 — APPROVED** [review]

File: src/together/resources/finetune.py (hunk 1 above)
Comment (azahed98) [reply src/together/resources/finetune.py:163]: Oops ty for the catch. I'm actually not sure it didn't have a validation error before...
